Metalama.Framework.Engine/Pipeline/DesignTime/DesignTimeSyntaxTreeGenerator.cs

```diff
@@ -153,6 +172,120 @@ void ProcessTransformationsOnType( IGrouping<INamedType, ITransformation> transf
             return additionalSyntaxTreeDictionary.Values.AsReadOnly();
         }
 
+        private static IReadOnlyList<ConstructorDeclarationSyntax> CreateInjectedConstructors(
+            CompilationModel initialCompilationModel,
+            CompilationModel finalCompilationModel,
+            SyntaxGenerationContext syntaxGenerationContext,
+            Dictionary<IConstructor, List<IParameter>> introducedParameterMap )
+        {
+            // TODO: This will not work properly with universal constructor builders.
+
+            var constructors = new List<ConstructorDeclarationSyntax>();
+            var signatures = new Dictionary<INamedType, HashSet<(ISymbol Type, Code.RefKind RefKind)[]>>( finalCompilationModel.Comparers.Default );
+
+            // Go through all types that will get generated constructors and index existing constructors.
+            var comparer = new ConstructorSignatureEqualityComparer();
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            foreach ( var type in introducedParameterMap.Keys.Select( c => c.DeclaringType ).Distinct( finalCompilationModel.Comparers.Default ) )
```

Comment:
This is processing constructors for a single type, right? So how can there be multiple `DeclaringType`s?

Reply:
Yeah, leftover.